fastapi/routing.py

```diff
@@ -67,15 +67,17 @@ def _prepare_response_content(
     exclude_unset: bool,
     exclude_defaults: bool = False,
     exclude_none: bool = False,
+    field: Optional[ModelField] = None,
 ) -> Any:
-    if isinstance(res, BaseModel):
-        read_with_orm_mode = getattr(res.__config__, "read_with_orm_mode", None)
+    if field is not None and issubclass(field.type_, BaseModel):
```

Comment:
I think you need to use `issubclass(type(field.type_), BaseModel)` or just use `isinstance`. Also, it would be nice if you add some tests

Reply:
no, `field.type_` is the class passed via `response_model`, to check if it supports validation with orm mode we need to check if it is a subclass of `BaseModel`.